apps/openmw/mwmechanics/aicombat.cpp

```diff
@@ -231,7 +232,13 @@ namespace MWMechanics
             storage.stopAttack();
             characterController.setAttackingOrSpell(false);
             storage.mActionCooldown = 0.f;
-            forceFlee = true;
+            if (target == MWMechanics::getPlayer())
+                forceFlee = true;
+            else
+            {
+                CreatureStats& creatureStats = actor.getClass().getCreatureStats(actor);
+                creatureStats.getAiSequence().stopCombat();
```

Comment:
This sounds very dangerous to do while the package is still being processed. You should have AiPackage::execute() return true if you want it to be removed correctly.